src/app/main/employees-sheet/sheet-form/sheet-form.component.ts

```diff
@@ -14,31 +18,93 @@ import { SheetsListComponent } from '../sheets-list/sheets-list.component';
 export class SheetFormComponent implements OnInit {
   @ViewChild('attendanceSheetForm') attendanceSheetForm: NgForm;
   private subscriptions: Subscription[] = [];
-  holidays: any[] = []
+  holidays: any[] = [];
+  isFileLoading: boolean = false;
+  isFileHasError: boolean = false;
+  isFileSuccess: boolean = false;
+  isFileUpdating: boolean = false;
+  nationalHolidayDate: any;
+  sheetFile: any;
+  sheet: AttendanceSheet = new AttendanceSheet();
   constructor(
     public dialogRef: MatDialogRef<SheetsListComponent>,
     @Inject(MAT_DIALOG_DATA) public data: any,
     private appNotificationService: AppNotificationService,
     public translate: TranslateService,
-    private dialog: MatDialog
+    private dialog: MatDialog,
+    private employeesSerivce: EmployeesSerivce,
+    private uploadFilesWithPreSignedUrlService: UploadFilesWithPreSignedUrlService
+
   ) { }
 
   ngOnInit(): void {
+    //  this.sheet = new AttendanceSheet();
 
   }
- 
+
   public closeModal() {
     this.dialogRef.close();
   }
-  addHoliday(holiday:any){
-
-console.log(holiday);
-this.holidays.push(holiday)
+  addHoliday(holidayDate: any) {
+    holidayDate = moment(holidayDate).format('YYYY-MM-DD');
+    this.nationalHolidayDate = ''
+    if(this.sheet.national_holidays.filter(nationalDay=> nationalDay.day.toString() == holidayDate.toString() ).length > 0){
```

Comment:
Could you please create a separate method for this condition (just to be more descriptive)

Reply:
ok, done